android/app/src/main/kotlin/com/elodin/walkie_talkie/HostAdvertiser.kt

```
@@ -0,0 +1,324 @@
+        }
+
+        // Setting setIncludeDeviceName(true) below pulls the *adapter* name
+        // into the adv frame; that's a system-wide setting that survives
+        // process death. Capture the current value so stop() can put it
+        // back — otherwise a Frequency host session would permanently
+        // rename the user's phone in their Bluetooth settings. Skipping
+        // the rename if it already matches avoids touching the adapter
+        // when the user's existing BT name is already their display name.
+        val previousName = try {
+            adapter.name
+        } catch (e: SecurityException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)